### PR TITLE
[v2.0] [BC Break] Rewrite library with ConsoleHelper + tests

### PR DESCRIPTION
**Pull request by michalbundyra**
Rewrite library with ConsoleHelper + tests.
**Rebase needed** after releasing v1.1.1 with fixes #5, #6, #7.

These changes provide new script:
```bash
$ vendor/bin/zf-composer-autoloading <command> [options] modulename
```

Commands:
  - `help`    - display help/usage message
  - `enable`  - enable composer autoloading for the module
  - `disable` - disbale composer autoloading for the module

Options are the same as previously:
  - `--help|-h`         - display help/usage message
  - `--composer|-c`     - composer binary path; defaults composer
  - `--type|-t`         - module autoloading type to use
  - `--modules-path|-p` - path to directory with modules

Also added full test coverage. **Please enable coveralls for the repository.**

### Note
This version will be used in https://github.com/zendframework/zend-expressive-tooling/pull/12 as it provides functionality also to _disable_ module's composer autoloading.

### CODE REVIEW

**Comment by michalbundyra** [comment]
Green! :) Ready for review @weierophinney

**Comment by michalbundyra** [comment]
@weierophinney rebased

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
         "console",
         "framework",
         "zendframework",
-        "zf2"
+        "zf3"
```
Comment (weierophinney) [inline composer.json:11]: I'd argue we should drop the version-specific tags, and just stick with `zendframework`.

File: src/Command.php [hunk 2]
```diff
@@ -6,203 +6,188 @@
...
-     * @var string $path
+     * @param string $command
+     * @param null|ConsoleHelper $console
      */
-    public function __construct($path)
+    public function __construct($command = self::DEFAULT_COMMAND_NAME, ConsoleHelper $console = null)
     {
-        $this->path = $path;
-        $this->composerJsonFile = sprintf('%s/composer.json', $path);
+        $this->command = (string)$command;
```
Comment (weierophinney) [inline src/Command.php:64]: Should be a space following the `)`; I thought that was setup in the zend-coding-standard?

**Review by weierophinney — APPROVED** [review]
Two minor nitpicks; overall, love this. I'll take care of the requested changes on merge.

**Comment by weierophinney** [comment]
Thanks a ton for this, @webimpress\!

File: src/Command.php (hunk 2 above)
Comment (michalbundyra) [reply src/Command.php:64]: Yeh, there should be space. It is not in version 1.0.0 of zend-coding-standard unfortunately but it is coming with big upgrade ;-)

**Pull request merged**